docs/rollover.adoc

```diff
@@ -0,0 +1,65 @@
+
+== `rollover.hpp`
+
+https://github.com/intel/cpp-std-extensions/blob/main/include/stdx/rollover.hpp[`rollover.hpp`]
+provides a class template `rollover_t` that is intended to act like an unsigned
+integral type, but with semantics that include the ability to roll-over on overflow.
+
+A `rollover_t` can be instantiated with any unsigned integral type:
+
+[source,cpp]
+----
+// explicit type
+auto x = stdx::rollover_t<std::uint8_t>{};
+
+// deduced type is equivalent to the unsigned version of the argument
+auto y = stdx::rollover_t{1}; // rollover_t<unsigned int>
+----
+
+It supports all the usual arithmetic operations (`+` `-` `*` `/` `%`) and
+behaves much like an unsigned integral type, with defined overflow and underflow
+semantics. The difference is in the comparison operators. A `rollover_t`
+considers itself to be in the middle of a rolling window where half the
+bit-range is always lower and half is higher.
+
+For instance, imagine a 3-bit unsigned integral type. There are eight values of
+this type: `0` `1` `2` `3` `4` `5` `6` `7`. For any value, there are always four
+values less than it, and three values greater than it. (And of course it is equal to
+itself, the eighth value). e.g. for the value `5`:
+
+- `1`, `2`, `3`, `4` are all less than `5`
+- `6`, `7`, `0` are all greater than `5`
+
+Effectively any value partitions the cyclic space into (almost) equal parts.
+
+CAUTION: `operator<` on `rollover_t` is not antisymmetric!
+
+This has some interesting consequences, for example both of the following are true:
+
+- `1` < `5`
+- but also, `5` < `1`! (The four values less than `1` are `5`, `6`, `7`, `0`)
+
+CAUTION: `operator<` on `rollover_t` is not transitive!
+
+Also, the following are all true:
+
+- `1` < `3`
+- `3` < `5`
+- `5` < `7`
+- `7` < `1`
```

Comment:
comment: it makes sense to me that it is not transitive....i can see that we can have multiple comparisons where the distance between each pair is less than half the range and so we can wrap completely around with a few pair-wise comparisons.